sql: panic: EXPORT INTO CSV _ FROM TABLE _

```
CREATE TABLE table1 (col1_0 BYTES, col1_1 REGPROC);

ALTER TABLE table1
        INJECT STATISTICS e'[{"name":"__auto__","created_at":"2000-01-01 00:00:00+00:00","columns":["col1_0"],"row_count":8249602943838988474,"distinct_count":6498702402579497593,"null_count":0,"histo_col_type":"BYTES","histo_buckets":[{"num_eq":9000000000,"num_range":0,"distinct_range":0,"upper_bound":"\\\\x22"},{"num_eq":30000000000,"num_range":3127288617296586065,"distinct_range":3127288617296586000,"upper_bound":"\\\\x526cbf0794315e69"},{"num_eq":1000,"num_range":1917088775817291864,"distinct_range":1917088775817291800,"upper_bound":"\\\\xc386b322d4e4"},{"num_eq":5091004749801871085,"num_range":200000,"distinct_range":189842.41629062488,"upper_bound":"\\\\xeeaa10e74af8a1f982"}]},{"name":"__auto__","created_at":"2000-01-01 00:00:00+00:00","columns":["col1_1"],"row_count":8249602943838988474,"distinct_count":3687336393122262357,"null_count":0,"histo_col_type":"REGPROC","histo_buckets":[{"num_eq":7974423192766406779,"num_range":0,"distinct_range":0,"upper_bound":"394377767"},{"num_eq":0,"num_range":800,"distinct_range":0,"upper_bound":"1484705942"},{"num_eq":704977322447399860,"num_range":771897481019995201,"distinct_range":0,"upper_bound":"1603303425"},{"num_eq":30000000000,"num_range":100,"distinct_range":100,"upper_bound":"3140507175"},{"num_eq":463531236131036400,"num_range":192637691944801783,"distinct_range":192637691944801800,"upper_bound":"3161349294"}]}]';

EXPORT
INTO CSV
        'http://0.1:37957/exp_1'
FROM
        TABLE table1;
```

```
* ERROR: [n1,client=127.0.0.1:45820,user=root] a SQL panic has occurred while executing "EXPORT INTO CSV 'http://0.1:37957/exp_1' FROM TABLE table1": runtime error: invalid memory address or nil pointer dereference
*
*
* ERROR: [n1,client=127.0.0.1:45820,user=root] a panic has occurred!
*
panic while executing 1 statements: EXPORT INTO CSV _ FROM TABLE _

goroutine 1252 [running]:
runtime/debug.Stack(0x48455a0, 0xc0009a6c00, 0x3)
        /usr/local/go/src/runtime/debug/stack.go:24 +0x9d
github.com/cockroachdb/cockroach/pkg/util/log.ReportPanic(0x48455a0, 0xc0009a6c00, 0xc000abe000, 0x3e2ab40, 0xc0010c94d0, 0x1)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/util/log/crash_reporting.go:217 +0xb5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).closeWrapper(0xc00157e000, 0x48455a0, 0xc0009a6c00, 0x3b2b2a0, 0x699ef70)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:736 +0x348
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn.func1(0xc00157e000, 0x48455a0, 0xc0009a6c00)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:449 +0x61
panic(0x3b2b2a0, 0x699ef70)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/cockroachdb/cockroach/pkg/sql/colexec/execerror.CatchVectorizedRuntimeError.func1(0xc000b0f970)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/execerror/error.go:82 +0x5d0
panic(0x3b2b2a0, 0x699ef70)
        /usr/local/go/src/runtime/panic.go:679 +0x1b2
sync.(*Mutex).Lock(...)
        /usr/local/go/src/sync/mutex.go:74
github.com/cockroachdb/cockroach/pkg/base.(*ClusterIDContainer).Get(0x0, 0x0, 0x0)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/base/cluster_id.go:46 +0x35
github.com/cockroachdb/cockroach/pkg/ccl/importccl.newCSVWriterProcessor(0xc0017c5500, 0x1, 0xc000b26fad, 0x16, 0x40180ec, 0xa, 0x0, 0x0, 0x0, 0x186a0, ...)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/ccl/importccl/exportcsv.go:44 +0x4b
github.com/cockroachdb/cockroach/pkg/sql/rowexec.NewProcessor(0x4845660, 0xc0018204b0, 0xc0017c5500, 0x1, 0xc001362d90, 0xc001362e58, 0xc000388310, 0x1, 0x1, 0xc000388370, ...)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/rowexec/processors.go:243 +0x134c
github.com/cockroachdb/cockroach/pkg/sql/colexec.NewColOperator.func1(0xc000388310, 0x1, 0x1, 0x0, 0x1, 0xc000388310, 0x0)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/execplan.go:423 +0x158
github.com/cockroachdb/cockroach/pkg/sql/colexec.wrapRowSources(0x4845660, 0xc0018204b0, 0xc0017c5500, 0xc00037dac0, 0x1, 0x2, 0xc00037dca0, 0x1, 0x1, 0xc00037dbc0, ...)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/execplan.go:81 +0xf9
github.com/cockroachdb/cockroach/pkg/sql/colexec.NewColOperator(0x4845660, 0xc0018204b0, 0xc0017c5500, 0xc001362d78, 0xc00037dac0, 0x1, 0x2, 0xc00037dbc0, 0x0, 0x41a78a0, ...)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/execplan.go:412 +0x52f3
github.com/cockroachdb/cockroach/pkg/sql/colflow.(*vectorizedFlowCreator).setupFlow(0xc001273e00, 0x4845660, 0xc0018204b0, 0xc0017c5500, 0xc001362c00, 0x2, 0x2, 0x1, 0x98, 0xc00122bcc0, ...)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:717 +0x859
github.com/cockroachdb/cockroach/pkg/sql/colflow.SupportsVectorized.func2()
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:960 +0x7e
github.com/cockroachdb/cockroach/pkg/sql/colexec/execerror.CatchVectorizedRuntimeError(0xc000b0fa90, 0x0, 0x0)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/execerror/error.go:91 +0x5f
github.com/cockroachdb/cockroach/pkg/sql/colflow.SupportsVectorized(0x4845660, 0xc0018204b0, 0xc0017c5500, 0xc001362c00, 0x2, 0x2, 0x1, 0x0, 0x0, 0x0, ...)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/colflow/vectorized_flow.go:959 +0x594
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).setupFlows(0xc0008b2780, 0x4845660, 0xc0018204b0, 0xc00157e470, 0xc0011fb540, 0xc00072dda0, 0xc0005ef600, 0xc00157e470, 0x1, 0xc001a7bb90, ...)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:165 +0x38e
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).Run(0xc0008b2780, 0xc0017c54a0, 0xc001a7bb90, 0xc000b105c0, 0xc0005ef600, 0xc00157e470, 0x0, 0x0)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:353 +0x375
github.com/cockroachdb/cockroach/pkg/sql.(*DistSQLPlanner).PlanAndRun(0xc0008b2780, 0x4845660, 0xc0018204b0, 0xc00157e470, 0xc0017c54a0, 0xc001a7bb90, 0x48471e0, 0xc0017c53e0, 0xc0005ef600, 0xc00157e1d8)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/distsql_running.go:976 +0x209
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execWithDistSQLEngine(0xc00157e000, 0x4845660, 0xc0018204b0, 0xc00157e390, 0x3, 0x7fddf472b538, 0xc001caa680, 0xc000b10901, 0x0, 0x0, ...)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:823 +0x3b1
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).dispatchToExecutionEngine(0xc00157e000, 0x4845660, 0xc0018204b0, 0xc00157e390, 0x7fddf472b538, 0xc001caa680, 0x0, 0x0)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:715 +0x6e8
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmtInOpenState(0xc00157e000, 0x4845660, 0xc0018204b0, 0x485c9e0, 0xc000858b40, 0xc001c9d5b4, 0x4a, 0x0, 0x1, 0x0, ...)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:418 +0xc30
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execStmt(0xc00157e000, 0x4845660, 0xc0018204b0, 0x485c9e0, 0xc000858b40, 0xc001c9d5b4, 0x4a, 0x0, 0x1, 0x0, ...)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor_exec.go:100 +0x4f6
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).execCmd(0xc00157e000, 0x48455a0, 0xc000858d00, 0x0, 0x0)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1283 +0x1ab5
github.com/cockroachdb/cockroach/pkg/sql.(*connExecutor).run(0xc00157e000, 0x48455a0, 0xc0009a6c00, 0xc0000739c0, 0x5400, 0x15000, 0xc000073a58, 0xc000328f10, 0x0, 0x0)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:1212 +0x1f2
github.com/cockroachdb/cockroach/pkg/sql.(*Server).ServeConn(0xc000552a00, 0x48455a0, 0xc0009a6c00, 0xc00157e000, 0x5400, 0x15000, 0xc000073a58, 0xc000328f10, 0x0, 0x0)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/conn_executor.go:451 +0x104
github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync.func1(0xc001a3893f, 0xc0018d90c0, 0x48455a0, 0xc0009a6c00, 0xc000328f10, 0xc000552a00, 0xc001caa000, 0x485b520, 0xc0018d90a0, 0xc000125860, ...)
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:566 +0x2e8
created by github.com/cockroachdb/cockroach/pkg/sql/pgwire.(*conn).processCommandsAsync
        /home/mjibson/go/src/github.com/cockroachdb/cockroach/pkg/sql/pgwire/conn.go:498 +0x17e

panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: runtime error: invalid memory address or nil pointer dereference [recovered]
        panic: panic while executing 1 statements: EXPORT INTO CSV _ FROM TABLE _; caused by runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x121ce35]
```

--- a/pkg/ccl/importccl/exportcsv_test.go
+++ b/pkg/ccl/importccl/exportcsv_test.go
@@ -217,3 +217,19 @@ func TestExportShow(t *testing.T) {
 		t.Fatalf("expected %q, got %q", expected, got)
 	}
 }
+
+// TestExportVectorized makes sure that SupportsVectorized check doesn't panic
+// on CSVWriter processor.
+func TestExportVectorized(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	dir, cleanupDir := testutils.TempDir(t)
+	defer cleanupDir()
+
+	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{ExternalIODir: dir})
+	defer srv.Stopper().Stop(context.Background())
+	sqlDB := sqlutils.MakeSQLRunner(db)
+
+	sqlDB.Exec(t, `CREATE TABLE t(a INT PRIMARY KEY)`)
+	sqlDB.Exec(t, `SET vectorize_row_count_threshold=0`)
+	sqlDB.Exec(t, `EXPORT INTO CSV 'http://0.1:37957/exp_1' FROM TABLE t`)
+}

--- a/pkg/sql/distsql_running.go
+++ b/pkg/sql/distsql_running.go
@@ -168,6 +168,7 @@ func (dsp *DistSQLPlanner) setupFlows(
 						Cfg: &execinfra.ServerConfig{
 							DiskMonitor: &mon.BytesMonitor{},
 							Settings:    dsp.st,
+							ClusterID:   &dsp.rpcCtx.ClusterID,
 						},
 						NodeID: -1,
 					}, spec.Processors, fuseOpt,

--- a/pkg/sql/explain_plan.go
+++ b/pkg/sql/explain_plan.go
@@ -223,6 +223,7 @@ func (p *planner) populateExplain(
 		distSQLPlanner.FinalizePlan(planCtx, &physicalPlan)
 		flows := physicalPlan.GenerateFlowSpecs(params.extendedEvalCtx.NodeID)
 		flowCtx := makeFlowCtx(planCtx, physicalPlan, params)
+		flowCtx.Cfg.ClusterID = &distSQLPlanner.rpcCtx.ClusterID
 
 		ctxSessionData := flowCtx.EvalCtx.SessionData
 		vectorizedThresholdMet := physicalPlan.MaxEstimatedRowCount >= ctxSessionData.VectorizeRowCountThreshold

--- a/pkg/sql/explain_vec.go
+++ b/pkg/sql/explain_vec.go
@@ -72,6 +72,7 @@ func (n *explainVecNode) startExec(params runParams) error {
 	distSQLPlanner.FinalizePlan(planCtx, &plan)
 	flows := plan.GenerateFlowSpecs(params.extendedEvalCtx.NodeID)
 	flowCtx := makeFlowCtx(planCtx, plan, params)
+	flowCtx.Cfg.ClusterID = &distSQLPlanner.rpcCtx.ClusterID
 
 	// Temporarily set vectorize to on so that we can get the whole plan back even
 	// if we wouldn't support it due to lack of streaming.

